### PR TITLE
add GetHistoryByLabelID interface with unit test, Register history's UrlPatterns to chassis

### PR DESCRIPTION
**Pull request by liuqi04**
1. add getHistoryByLabelID interface along with unit test. also modify some existing interfaces and test since we have updated go-restful for api tests
2. new interface fit new dao structure
3. modify some doc fields for history interfaces implementation

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,15 +4,20 @@ require (
...
 	github.com/onsi/gomega v1.5.0
+	github.com/pkg/errors v0.8.1
 	github.com/stretchr/testify v1.3.0
+	github.com/tidwall/pretty v1.0.0 // indirect
+	github.com/uber-go/atomic v1.4.0 // indirect
 	github.com/urfave/cli v1.20.0
 	github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c // indirect
 	github.com/xdg/stringprep v1.0.0 // indirect
-	go.mongodb.org/mongo-driver v1.0.3
+	go.mongodb.org/mongo-driver v1.0.0
```
Comment (tianxiaoliang) [inline go.mod]: change to latest version

File: pkg/model/mongodb_doc.go [hunk 2]
```diff
@@ -45,8 +45,8 @@ type KVDoc struct {
 type LabelRevisionDoc struct {
 	ID       primitive.ObjectID `json:"_id,omitempty" bson:"_id,omitempty"`
 	LabelID  string             `json:"label_id,omitempty"  bson:"label_id,omitempty"`
-	Labels   map[string]string  `json:"labels,omitempty"` //redundant
-	Domain   string             `json:"domain,omitempty"` //redundant
-	KVs      []*KVDoc           `json:"data,omitempty"`   // save states of this revision
+	Labels   map[string]string  `json:"labels,omitempty"`
+	Domain   string             `json:"domain,omitempty" bson:"-"`
```
Comment (tianxiaoliang) [inline pkg/model/mongodb_doc.go]: do not ignore bson, ignore json

File: server/resource/v1/history_resource.go [hunk 3]
```diff
@@ -17,6 +17,70 @@
...
+			ResourceFuncName: "GetRevisionsByLabelID",
+			FuncDesc:         "get all revisions by label id",
+			Parameters: []*restful.Parameters{
+				DocPathLabelID,
+			},
+			Returns: []*restful.Returns{
+				{
+					Code:    http.StatusOK,
+					Message: "true",
+					Model:   []*model.LabelHistoryResponse{},
```
Comment (tianxiaoliang) [inline server/resource/v1/history_resource.go]: change []*model.LabelHistoryResponse{} to []model.LabelHistoryResponse{}

using pointer will also generate a symbol * in yaml file

File: server/resource/v1/history_resource_test.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+	v1 "github.com/apache/servicecomb-kie/server/resource/v1"
+	"github.com/go-chassis/go-chassis/server/restful/restfultest"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+type FakeHandler struct{}
+
+func (fh *FakeHandler) Handle(chain *handler.Chain, i *invocation.Invocation, cb invocation.ResponseCallBack) {
```
Comment (tianxiaoliang) [inline server/resource/v1/history_resource_test.go]: you don't need to write  this fake handler, just use no op handler

File: server/resource/v1/history_resource_test.go [hunk 5]
```diff
@@ -81,8 +64,9 @@ var _ = Describe("v1 history resource", func() {
 			path := fmt.Sprintf("/v1/revision/%s", kv.LabelID)
 			r, _ := http.NewRequest("GET", path, nil)
 			revision := &v1.HistoryResource{}
-			handler.RegisterHandler("fake", newFakeHandler)
-			chain, _ := handler.CreateChain(common.Provider, "testRevisions", "fake")
+			// handler.RegisterHandler("fake", newFakeHandler)
```
Comment (tianxiaoliang) [inline server/resource/v1/history_resource_test.go]: never comment code,just remove

File: server/resource/v1/kv_resource_test.go [hunk 6]
```diff
@@ -33,6 +33,7 @@ import (
 	"github.com/apache/servicecomb-kie/pkg/model"
 	"github.com/apache/servicecomb-kie/server/config"
 	"github.com/apache/servicecomb-kie/server/db"
+	noop "github.com/apache/servicecomb-kie/server/handler"
```
Comment (tianxiaoliang) [inline server/resource/v1/kv_resource_test.go]: do not rename

File: server/resource/v1/history_resource_test.go (hunk 5 above)
Comment (liuqi04) [reply server/resource/v1/history_resource_test.go]: got it

**Pull request merged**